packages/common/src/store/buy-usdc/slice.ts

```diff
@@ -0,0 +1,86 @@
+import { Action, createSlice, PayloadAction } from '@reduxjs/toolkit'
+
+import { BuyUSDCStage, USDCOnRampProvider, PurchaseInfo } from './types'
+
+type StripeSessionStatus =
+  | 'initialized'
+  | 'rejected'
+  | 'requires_payment'
+  | 'fulfillment_processing'
+  | 'fulfillment_complete'
+
+type OnSuccess = {
+  action?: Action
+  message?: string
+}
+
+type BuyUSDCState = {
+  stage: BuyUSDCStage
+  error?: Error
+  provider: USDCOnRampProvider
+  onSuccess?: OnSuccess
+  stripeSessionStatus?: StripeSessionStatus
+}
+
+const initialState: BuyUSDCState = {
+  provider: USDCOnRampProvider.UNKNOWN,
+  stage: BuyUSDCStage.START
+}
+
+const slice = createSlice({
+  name: 'buy-usdc',
+  initialState,
+  reducers: {
+    startBuyUSDCFlow: (
+      state,
+      action: PayloadAction<{
+        purchaseInfo: PurchaseInfo
+        provider: USDCOnRampProvider
+        onSuccess?: OnSuccess
+      }>
+    ) => {
+      state.stage = BuyUSDCStage.START
+      state.error = undefined
+      state.provider = action.payload.provider
+      state.onSuccess = action.payload.onSuccess
+    },
+    onRampOpened: (state, _action: PayloadAction<PurchaseInfo>) => {
+      state.stage = BuyUSDCStage.PURCHASING
+    },
+    onRampCanceled: (state) => {
+      if (state.stage === BuyUSDCStage.PURCHASING) {
+        state.error = new Error('USDC purchase canceled')
```

Comment:
imo I don't consider cancelling an error

Reply:
errr, yeah I agree. I may have been thinking in the context of the content purchasing flow, where we can't complete your purchase if you cancel the buy usdc modal. But I should probably handle the cancelation there and set an error in that slice.

Reply:
Removed